### ISSUE TITLE
[naming-convention] modifier export - Allow selecting only exported members

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#2239]
**Repro**

```JSONC
{
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      { "selector": "variable", "format": ["camelCase"] },
      // { "selector": "variable", "modifiers": ["export"], "format": ["UPPER_CASE"] }
    ]
  }
}
```

```TS
export const FOO = 'hello';
const bar = 'world';
```

**Expected Result**

The ability to enforce a naming convention that only applies to exported members.

**Actual Result**

Cannot narrow down selector enough.

**Additional Info**

Related to: #1612

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.4.0` |
| `@typescript-eslint/parser`        | `3.4.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `14.4.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
how come you want to specifically name exported variables differently to non-exported variables?

**Comment by RebeccaStevens** [comment]
When importing constants, I tend to like them being in upper case so that I can tell they are a constant at a glance and not something else like a function.

When just using constants in a module (without exporting them), there's no need to make upper case as it is clear from context what they are.

**Comment by Mossop** [comment]
I came to suggest the same thing but because I do things the other way around. I tend to use constants within a module for magic numbers and such but for exported objects prefer camel case.

**Comment by nicolas-cherel** [comment]
> how come you want to specifically name exported variables differently to non-exported variables?

Looks to me that it is the reasons for public/private class members applies really well for exported and non exported identifiers within a module

**Comment by nicolas-cherel** [comment]
I'd also add this modifier for `typeLike` selector
